app/controllers/writable_controller.rb

```diff
@@ -13,8 +13,8 @@ def build_template_groups(user=nil)
     @templates = templates + [templateless]
 
     if @post
-      uniq_chars_ids = @post.replies.where(user_id: user.id).where('character_id is not null').group(:character_id).pluck(:character_id)
-      uniq_chars_ids << @post.character_id if @post.user_id == user.id && @post.character_id.present?
+      uniq_chars_ids = @post.replies.where(user: user).select(:character_id).distinct.pluck(:character_id)
+      uniq_chars_ids << @reply.character_id if @reply.present?
```

Comment:
Why were we checking user on the local reply before? was it for coauthors or mod edits?